lib/srv/server/ssm_install.go

```diff
@@ -268,19 +274,33 @@ func (si *SSMInstaller) emitInvalidInstanceEvents(ctx context.Context, req SSMRu
 
 // instanceIDsSSMState contains a list of EC2 Instance IDs for a given state.
 type instanceIDsSSMState struct {
-	valid          []string
-	missing        []string
-	connectionLost []string
-	unsupportedOS  []string
+	valid          map[string]string
+	missing        map[string]string
+	connectionLost map[string]string
+	unsupportedOS  map[string]string
+}
+
+func instanceIDsFrom(m map[string]string) []string {
+	ret := make([]string, 0, len(m))
+	for k := range m {
+		ret = append(ret, k)
+	}
+	return ret
```

Comment:
The go stdlib now has a way of doing this, but I'm not really sure it's more readable:

```
import (
    "maps"
    "slices"
)
...
        return slices.Collect(maps.Keys(m))
```

That's not identical as it does not pre-allocate a slice of the correct size. You could do that with:
```
        return slices.AppendSeq(make([]string, 0, len(m)), maps.Keys(m))
```
... and this is where it becomes more unreadable.

The package `golang.org/x/exp/maps` had `maps.Keys(m)` that returned a slice, but that did not make it into the go stdlib since we got range functions, so I don't think its a good idea to use `x/exp/maps`.

I don't feel strongly about this, just mentioning it if you prefer to use stdlib.

Reply:
TIL, thank you.
Updated to your suggestion